insights/specs/default.py

```diff
@@ -710,6 +712,34 @@ def pcp_enabled(broker):
     pcs_status = simple_command("/usr/sbin/pcs status")
     php_ini = first_file(["/etc/opt/rh/php73/php.ini", "/etc/opt/rh/php72/php.ini", "/etc/php.ini"])
     pluginconf_d = glob_file("/etc/yum/pluginconf.d/*.conf")
+
+    @datasource(Ps, Hostname, HostContext)
+    def pmlog_summary(broker):
+        """
+        Returns the output of the pmlogsummary command if a running ``pmlogger``
+        process is detected on the system.
+
+        Returns:
+            DatasourceProvider: contains the output of the command
+
+        Raises:
+            SkipComponent: raises this exception when the command is not present or fails
+        """
+        ps = broker[Ps]
+        hostname = broker[Hostname].fqdn
+        if ps.search(COMMAND__contains='pmlogger'):
+            pcp_log_date = (datetime.date.today() - datetime.timedelta(days=1)).strftime("%Y%m%d")
+            try:
+                cmd = "/usr/bin/pmlogsummary /var/log/pcp/pmlogger/%s/%s.index mem.util.used mem.physmem kernel.all.cpu.user kernel.all.cpu.sys kernel.all.cpu.nice kernel.all.cpu.steal kernel.all.cpu.idle disk.all.total" % (hostname, pcp_log_date)
```

Comment:
I'd create the datasource as a helper that constructs the file path and checks that it exists. Return it if so but raise the Skip otherwise. Then use [command_with_args](https://github.com/RedHatInsights/insights-core/blob/master/insights/core/spec_factory.py#L751) to interpolate it into the pmlogsummary command. That'll make sure the command gets timeouts applied, exceptions are consistent, etc.

Reply:
Done